app/assets/purs/src/EntryPoints/Comments.purs

```diff
@@ -0,0 +1,106 @@
+module EntryPoints.Comments
+  ( boot
+  ) where
+
+import Prelude
+
+import Data.Array as Array
+import Data.Maybe (Maybe(..))
+import Elmish (BootRecord, ComponentDef, Dispatch, ReactElement, Transition, handleMaybe)
+import Elmish.Foreign (Foreign, readForeign)
+import Elmish.HTML.Styled as H
+import Foreign.Object as Obj
+import Utils.Boot (bootOrPanic)
+-- import Utils.API as API
+
+boot :: BootRecord Foreign
+boot = bootOrPanic
+  { def: Just <<< def
+  , diagnosticName: "Demo"
+  }
+
+data Message
+  = Create
+  | CommentTextChanged String
+  | CommenterChanged String
+
+type Props =
+  { comments :: Array Comment
+  }
+
+type Comment = 
+  { commenter :: String 
+  , comment :: String 
+  }
+
+type State = 
+  { commenter :: String
+  , newComment :: String
+  , comments :: Array Comment
+  }
+
+def :: Props -> ComponentDef Message State
+def props =
+  { init, update, view }
+  where
+    init :: Transition Message State
+    init = 
+      pure
+        { newComment: ""
+        , commenter : ""
+        , comments: props.comments }
+
+
+    update :: State -> Message -> Transition Message State
+    update state = case _ of
+      CommentTextChanged updatedComment ->
+        pure state 
+          { newComment = updatedComment }
```

Comment:
Not going to lie, I have no idea what this is. I know it's not a Record since that syntax uses `{key: value}`. `{ foo = bar}`

Reply:
It is syntax that is pretty much saying "Keep all the fields on `state` the same except for what we define within the curly braces.  So you had a record like 
```
 type Person = { name :: String, age :: Int }
```
the following would be equivalent
```
-- david :: Person
olderDavid = { name: david.name, age: david.age + 1 }
olderDavid = david { age = david.age + 1 }
```

Reply:
Ah, that makes sense. Thanks!